src/nix/flake.cc

```diff
@@ -25,6 +26,23 @@ using namespace nix;
 using namespace nix::flake;
 using json = nlohmann::json;
 
+class TypeMismatch : public Error
```

Comment:
```suggestion
/**
 * An expression accessed through the evaluation cache did not yield the desired result.
 */
class CachedResultMismatch : public Error
```

It's not only about types. Could be a cached error, for example.

Reply:
Maybe that shouldn't be, and `AttrCursor::showType` should just throw (or return `std::nullopt`) if the cached value isn't an actual value?